### ISSUE TITLE
Build pipeline and package build tasks

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#642]
These are all stretch goals for the 2.1 RC timeline.

- [x] if tests succeed, build packages in same job
- [x] archive .debs/.rpms at end of build phase
- [x] Nightly: build repository directory given a pile of .debs/.rpms
- [x] Nightly: push repos to couchdb-vm2
- [x] Release: push 2.0 deb/rpm binaries to bintray as validation
- [x] Release: build 2.0 snaps, push 2.0 snaps to snapstore
- [x] Release: Complete https://github.com/apache/couchdb-documentation/issues/141
- [x] Release: add GPG signing key to bintray and sign 2.0 packages
- [x] Release: update release process docs with pointers on deb/rpm/win/mac/docker/snap upload process
- [x] Release: for 2.1, build convenience binaries, test, publish!
- [ ] ~~Release: docker - relocate 2.0 image to apache/couchdb namespace (INFRA-14586)~~ tracked as apache/couchdb-docker#20
- [ ] ~~Release: Add Debian 9 (stretch) to the list of supported releases~~ https://github.com/William-Yeh/docker-ansible/pull/31 this has to land first :(

**Comment by wohali** [comment]
If all of this hits, also complete https://github.com/apache/couchdb-documentation/issues/141

**Comment by wohali** [comment]
We're auto-building repos/dist-type source tarballs on master now.

**Comment by wohali** [comment]
2.0 packages are on bintray.

https://apache.bintray.com/couchdb-rpm/
https://apache.bintray.com/couchdb-deb/

For a generated repo file: `wget https://bintray.com/apache/couchdb-rpm/rpm -O bintray-apache-couchdb-rpm.repo`

For debian, `echo "deb https://apache.bintray.com/couchdb-deb {distribution} main" | sudo tee -a /etc/apt/sources.list` where `{distribution}` is one of `jessie trusty xenial`.

**Comment by wohali** [comment]
Docs are updated, snaps are pushed.

**Comment by wohali** [comment]
Debian packages are signed. RPMs, well, are ugly. See https://serverfault.com/questions/624888/bad-signatures-or-nokey-errors-on-rpms-i-just-signed . Just ugh. We're not going to bother signing RPM packages for now.
